test/unit/foreman/renderer_deprecation_test.rb

```diff
@@ -0,0 +1,31 @@
+require 'test_helper'
+
+class RendererTest < ActiveSupport::TestCase
+  class DummyRenderer
+    attr_accessor :host
+
+    include Foreman::Renderer
+  end
+
+  test 'should show the deprecation for @host.info' do
+    renderer = DummyRenderer.new
+    renderer.host = FactoryGirl.create(:host, :with_puppet)
+    template = FactoryGirl.create(:provisioning_template, :template => '<%= @host.info %>')
+    logger = Rails.logger
+    Rails.logger = mock()
```

Comment:
A mock object isn't really required here, you can set the expectation on the existing Rails.logger. This removes the need to replace it and any errors from not restoring the original correctly (e.g. it's currently missing an ensure block).